### PR TITLE
Remove RadialBar branch from getItemByXY

### PR DESCRIPTION
**Pull request by PavelVanecek**
## Description

it's unreachable code for two reasons:

1. ~RadialBar does not have `activeBar` prop~ it does have `activeShape` prop! But it's still not reaching this code because of reason (2)
2. ~Even if I go ahead and add the `activeBar` prop (it doesn't do anything) and then silence it with typescript,~ it still doesn't reach this branch because activeTooltipIndex defaults to zero

Same story as Bar before: https://github.com/recharts/recharts/pull/4408

## Related Issue

https://github.com/recharts/recharts/discussions/3717

## Motivation and Context

Remove dead code

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes

### CODE REVIEW

**Comment by ckifer** [comment]
`RadialBar` has an `activeShape` prop

**Review by ckifer — CHANGES_REQUESTED** [review]
Looks good, we could probably rename the function though

File: src/chart/generateCategoricalChart.tsx [hunk 1]
```diff
@@ -1889,23 +1888,14 @@ export const generateCategoricalChart = ({
         ...this.state,
       });
 
-    public getItemByXY(chartXY: { x: number; y: number }) {
```
[the remark this answers is not in the document]
Comment (PavelVanecek) [reply src/chart/generateCategoricalChart.tsx:1892]: Renamed. I am going to remove this whole function eventually though.

**Comment by PavelVanecek** [comment]
I rebased and tested with `activeShape` too, thx @ckifer for bringing that up. All looks the same (activeShape boolean is not doing anything on 3.x and continues doing nothing here.)

**Comment by ckifer** [comment]
I mean it's not doing nothing, but it does not reach the removed code path

**Review by ckifer — APPROVED** [review]

**Pull request merged**